pkg/controllers/namespace/namespace_controller.go

```diff
@@ -206,16 +210,45 @@ func (c *NamespaceController) syncToCalico(key string) error {
 		return nil
 	} else {
 		// The object exists - update the datastore to reflect.
-		log.Infof("Add/Update Profile %s in Calico datastore", key)
+		log.Infof("Create/Update Profile %s in Calico datastore", key)
 		p := obj.(api.Profile)
-		_, err := c.calicoClient.Profiles().Apply(&p)
-		return err
+
+		// Lookup to see if this object already exists in the datastore.
+		gp, err := c.calicoClient.Profiles().Get(context.Background(), p.Name, options.GetOptions{})
+		if err != nil {
+			if _, ok := err.(errors.ErrorResourceDoesNotExist); !ok {
+				log.WithError(err).Warning("Failed to get profile from datastore")
+				return err
+			}
+
+			// Doesn't exist - create it.
+			_, err := c.calicoClient.Profiles().Create(context.Background(), &p, options.SetOptions{})
+			if err != nil {
+				log.WithError(err).Warning("Failed to create profile")
+				return err
+			}
+			log.Infof("Successfully created profile %s", key)
+			return nil
+		}
+
+		// The profile already exists - update it.
+		p.ResourceVersion = gp.ResourceVersion
+		log.Infof("Update Profile %s in Calico datastore with resource version %s", key, p.ResourceVersion)
+		_, err = c.calicoClient.Profiles().Update(context.Background(), &p, options.SetOptions{})
```

Comment:
shouldn't we have some sort of retry loop here if we get an update conflict error?

Reply:
There's actually a retry mechanism a level above this function - if it hits an error it will queue a retry.